internals/controllers/subnamespace_controller.go

```diff
@@ -201,6 +201,27 @@ func (r *SubnamespaceReconciler) Sync(ownerNamespace *utils.ObjectContext, subsp
 			}}
 		}
 	}
+
+	// trigger the child subnamespaces if the subnamespace was converted into a resourcepool
+	// it will update the isUpperRp annotation and the quotas accordingly
+	if utils.GetSnsResourcePooled(subspace.Object) == "true" {
+		for _, sns := range subspaceChilds.Objects.(*danav1.SubnamespaceList).Items {
+			snsChild, err := utils.NewObjectContext(subspace.Ctx, subspace.Log, subspace.Client, types.NamespacedName{Name: sns.GetName(), Namespace: subspace.GetName()}, &danav1.Subnamespace{})
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			isUpperResourcePool := snsChild.Object.GetAnnotations()[danav1.IsUpperRp]
+			if isUpperResourcePool == "True" {
+				r.SnsEvents <- event.GenericEvent{Object: &danav1.Subnamespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      snsChild.Object.GetName(),
+						Namespace: snsChild.Object.GetNamespace(),
+					},
+				}}
+			}
+		}
+	}
```

Comment:
The issue was that there were 2 or more upper `ResourcePools` under a `subnamespace` and then the `subnamespace` was turned into a `ResourcePool` then the former upper RPs were not updated correctly?

Reply:
exactly